postgres/rdspostgres/rdspostgres.go

```diff
@@ -97,9 +101,12 @@ func Open(ctx context.Context, provider rds.CertPoolProvider, params *Params) (*
 	vals := make(url.Values)
 	for k, v := range params.Values {
 		// Only permit parameters that do not conflict with other behavior.
-		if k == "sslmode" || k == "sslcert" || k == "sslkey" || k == "sslrootcert" {
```

Comment:
IIUC, these were exactly backwards; the ones that are allowable were listed, while the ones that conflict with `Params` were not. Should be the other way around.

Reply:
I don't agree with this change: I believe this to be correct as-is.

There was an issue in one iteration of this (across PRs) that wouldn't permit host in particular, which made it difficult to connect to databases using Unix domain sockets. I think for consistency we should allow those parameters, but we definitely want to forbid specifying any SSL/TLS options.

Reply:
Can you explain the intent then? IIUC, you can set the `user/password/dbname/host/port` fields:
* if you're using the explicit constructor, via `Params`
* if you're using a URL, via parts of the URL (which constructs a `Params`).

So why would you then want to also set those fields via a query parameter?
And, if we disallow query parameters like `sslmode` etc., how would you set them?

I interpreted the comment "Only permit parameters that do not conflict with other behavior" as "only permit parameters that don't conflict with what you can specify another way (via `Params`), but maybe I misunderstood.

Reply:
Per discussion, reverting this change. The SSL-related query parameters will definitely not work, because the underlying dialer provides TLS and so we explicitly disable SSL. I had to update the test, which was testing for the behavior I changed it to.

The query parameters like `user` etc. are probably not needed, because they are settable via `Params`, but there might be scenarios where they are useful (e.g., if you are using a URL opener and it can't parse your host correctly), so we don't gain much by disallowing them.